testing/utils/openshift_ci/pr_args.sh

```diff
@@ -89,17 +89,22 @@ else
     last_comment_page_json=$(curl -sSf "$PR_LAST_COMMENT_PAGE_URL")
 fi
 
-REQUIRED_AUTHOR_ASSOCIATION=CONTRIBUTOR
+REQUIRED_AUTHOUR=$(jq -r .user.login <<< "$pr_json")
+REQUIRED_AUTHOR_ASSOCIATION="CONTRIBUTOR"
 
 echo "PR comments URL: $PR_COMMENTS_URL" >&2
 last_user_test_comment=$(echo "$last_comment_page_json" \
-                             | jq '.[] | select(.author_association == "'$REQUIRED_AUTHOR_ASSOCIATION'") | .body' \
-                             | (grep "$test_anchor" || true) \
-                             | tail -1 | jq -r)
-
-if [[ -z "$last_user_test_comment" ]]; then
-    echo "ERROR: last comment of from a '$REQUIRED_AUTHOR_ASSOCIATION' could not be found (searching for '$test_anchor') ..." >&2
+                             | jq -r '.[-1] | .body | match("^/test.+") | .string')
```

Comment:
I think this: `jq -r '.[-1] | ....` will only look at the last comment of the PR, instead of searching for the last comment of the author or contributor

Reply:
the issue with that, that I originally wanted to avoid, is if someone comments after the test is launched, then the configuration will fail
So I prefer:
* search for the last comment that contains the test anchor
* check if this comment belongs to the author or a contributor

Reply:
hmmm the issue I see here is that, having `jq -r '.[-1] | ....` will never let `openshift-ci` run any jobs when a PR is given multiple comments at the same time, for example, one person comments with `/test <> <>` and other person comments with `<long-description-of-something-else>` but both comment at the same time. This situation will yield bad results with my change. So certainly we don't want this behavior. Let me fix it. 


>if someone comments after the test is launched, then the configuration will fail

I cannot think of any reason that causes this behavior. Can you tell me how ?

And also the existing regex `grep "$test_anchor"` considers the following comments as the potential candidates for the `openshift-ci` run. Please is this something do you wish to retain ? like searching whole comment for the test anchor.

```
[cmusali@cmusali topsail]$ curl -sSL https://api.github.com/repos/openshift-psap/topsail/issues/362/comments?per_page=100 | jq '.[] | select(.author_association == "CONTRIBUTOR") | .body' | (grep -E "/test .+" || true) | tail 
"Hello @mcharanrm , do you want to play again with this code?\r\nthe behavior I'd like to see is:\r\n1. find the last comment with `/test rhaoi-light` (the `$test_anchor`)\r\n2. check if this comment belongs to `$author` OR `.author_association == \"CONTRIBUTOR\"`\r\n3. if not, FAIL\r\n4. if yes, accept it an continue"

"> Hello @kpouget , I have pushed the changes and wanted to give it a try. So, I need some help here w.r.t the arguments that you are passing to the `kserve`.\r\n> \r\n> ```\r\n> test rhoai-plot kserve http://....\r\n> ```\r\n> \r\n> So what is `http://` pointing to ?\r\n\r\njust run it like that, actually I can do it :) \r\n\r\n/test rhoai-plot kserve http://...."

"/test rhoai-plot kserve http://...."
```

Reply:
> I cannot think of any reason that causes this behavior. Can you tell me how ?

yes, that's exactly what you describe at the top of the comment :)

> And also the existing regex grep "$test_anchor" considers the following comments as the potential candidates for the openshift-ci run.

yes, that's indeed an issue. `grep "^$test_anchor"` should avoid that. But overall, not a critical case

Reply:
I understand the problem now. I guess we can leave the regex pattern as it is now in order to trigger a openshift ci from a comment that also includes other insightful information.  I mean to say,  it doesn't necessarily have to be a separate comment to trigger the job. 

Looks like you got it everything covered in the existing code, like parsing the given comment and gathering the test anchor information

```
[cmusali@cmusali topsail]$ echo ' {"comment": "Lets try again on 4.15: \r\n\r\n/test rhoai-light kserve quick_raw_vllm" }' | jq '.comment' | grep -E '/test' | tail -1 | jq -r 
Lets try again on 4.15: 

/test rhoai-light kserve quick_raw_vllm

```

```
[cmusali@cmusali topsail]$ echo ' {"comment": "Lets try again on 4.15: \r\n\r\n/test rhoai-light kserve quick_raw_vllm" }' | jq '.comment' | grep -E '/test' | tail -1 | jq -r '' | (grep "/test" || true) | cut -d" " -f3- | tr -d '\n' | tr -d '\r'

kserve quick_raw_vllm

```